tealiumlibrary/src/main/java/com/tealium/core/Tealium.kt

```diff
@@ -282,10 +282,17 @@ class Tealium private constructor(val key: String, val config: TealiumConfig, pr
      * a replacement.
      */
     private fun getOrCreateVisitorId(): String {
-        return dataLayer.getString("tealium_visitor_id")
-                ?: UUID.randomUUID().toString().replace("-", "").also {
-                    dataLayer.putString("tealium_visitor_id", it, Expiry.FOREVER)
-                }
+        return dataLayer.getString("tealium_visitor_id") ?: run {
+             return config.existingVisitorId?.also {
+                 dataLayer.putString("tealium_visitor_id", it, Expiry.FOREVER)
+             } ?: ""//?: createVisitorId()
```

Comment:
still need the fallback to `createVisitorId` in the case where no `tealium_visitor_id` is set yet (1st launch) and the developer hasn't provided an `existingVisitorId` either.